GTHC/web
# OneSignal Integration

NOTE: Current seed users don't have netIDs. After running "Generate Shifts", in server logs, look for OneSignal notifications being created for your (the logged in user's) netid. 
Implemented Push Notifications for Shifts and Posts
1. Testing Notifications for Shifts
- Run ```yarn flush``` and ```yarn dev:start``` and make your own team 
- Enabling Notifications on Web: Navigate to http://localhost:5000/notif and click "Allow Notifications" in the browser pop up (rest assured this set up is temporary!) 
- Enabling Notifications on Mobile: Navigate to Notifications tab on app and select "Allow Notifications" in the pop up (waiting on updated build for mobile to be available). 
- Go to [http://localhost:5000/app/calendar](http://localhost:5000/app/calendar) and make a shift for yourself on the current day at a close future time 
- Verify that you receive a Notification about your shift 10 minutes before the shift's start time. 
2. Testing Notifications for Posts (Line Monitor Announcements) 
- Go to [http://localhost:5000/admin/posts](http://localhost:5000/admin/posts) and make a new post. 
- All subscribed users should receive a notification.
3. Testing Notifications Endpoint (Used to make our own Notifications History/Inbox)
- Be logged into GTHC with Shibboleth
- Navigate to [http://localhost:5000/api/v1/notifications/](http://localhost:5000/api/v1/notifications/) to get all (past and future) notifications for the current logged on user. The shift notifications can be filtered using their start_time attributes to get a notification history. 
- Use http://localhost:5000/api/v1/notifications/<user_id> to get the notifications user with that user_id. The shift notifications are under "shift_notifications" in the JSON, and Posts/Line Monitor announcements are under "announcements". 
Let Shamikh know if you encounter any issues during testing or if you don't get a notification. Test consciously and please be wary of spamming the OneSignal API with requests. All requests are logged in the server logs. If you see anything suspicious in the logs, please let me know. Thanks so much! :)
